docs/tutorials/demo_data_science_project.ipynb

```diff
@@ -184,8 +155,8 @@
    },
    "outputs": [],
    "source": [
-    "destination = \"data/weather-2010.json\"\n",
-    "urllib.request.urlretrieve(\"https://archive-api.open-meteo.com/v1/archive?latitude=52.52&longitude=13.41&start_date=2010-01-01&end_date=2010-12-31&hourly=temperature_2m,relativehumidity_2m,rain,pressure_msl,surface_pressure,cloudcover,cloudcover_low,cloudcover_mid,cloudcover_high,windspeed_10m,windspeed_100m,winddirection_10m,winddirection_100m\", destination)"
+    "outfile = tempfile.NamedTemporaryFile()\n",
+    "urllib.request.urlretrieve(\"https://archive-api.open-meteo.com/v1/archive?latitude=52.52&longitude=13.41&start_date=2010-01-01&end_date=2010-12-31&hourly=temperature_2m,relativehumidity_2m,rain,pressure_msl,surface_pressure,cloudcover,cloudcover_low,cloudcover_mid,cloudcover_high,windspeed_10m,windspeed_100m,winddirection_10m,winddirection_100m\", outfile.name)"
```

Comment:
After running this I got Permission error (not sure whether it is specific issue on my end):
---------------------------------------------------------------------------
PermissionError                           Traceback (most recent call last)
Cell In[8], line 2
      1 outfile = tempfile.NamedTemporaryFile()
----> 2 urllib.request.urlretrieve("https://archive-api.open-meteo.com/v1/archive?latitude=52.52&longitude=13.41&start_date=2010-01-01&end_date=2010-12-31&hourly=temperature_2m,relativehumidity_2m,rain,pressure_msl,surface_pressure,cloudcover,cloudcover_low,cloudcover_mid,cloudcover_high,windspeed_10m,windspeed_100m,winddirection_10m,winddirection_100m", outfile.name)

File ~\AppData\Local\Programs\Python\Python311\Lib\urllib\request.py:251, in urlretrieve(url, filename, reporthook, data)
    249 # Handle temporary file setup.
    250 if filename:
--> 251     tfp = open(filename, 'wb')
    252 else:
    253     tfp = tempfile.NamedTemporaryFile(delete=False)

PermissionError: [Errno 13] Permission denied: 'C:\\Users\\MACIEJ~1\\AppData\\Local\\Temp\\tmp4rroi7te'

Reply:
This is certainly Windows-specific, since it passes in CI. Some aspects of the `tempfile` module differ on Windows, so I imagine this is one of them (or Windows gives fewer default rights to the Temp location).

Reply:
This worked for me on Windows:
```
# Save fetched json as a temporary file
with tempfile.NamedTemporaryFile(delete=False, mode='w+', suffix='.json') as temp_file:
    temp_file.write(response.text)
    temp_file.seek(0)  # Move the file pointer to the beginning
```